

# Día 5 Parte 3: Carga de datos reales

## Ejemplos

**Ejemplo 5.12:**
 Carga de datos reales

In [2]:
import pandas as pd

tensiones = pd.read_csv('tensiones.csv', parse_dates=['fecha'], index_col='fecha')
print(tensiones.head())


                     tension1  tension2
fecha                                  
2021-01-01 00:15:00   119.549   118.911
2021-01-01 00:30:00   119.498   118.218
2021-01-01 00:45:00   118.792   118.595
2021-01-01 01:00:00   119.179   118.334
2021-01-01 01:15:00   119.298   118.063


**Ejemplo 5.13:** Promedio de una columna

In [ ]:
print(tensiones['tension1'].mean())

118.21804160528022


**Ejemplo 5.14:** Desviación estándar de columna

In [ ]:
print(tensiones['tension1'].std())

2.135782206081503


**Ejemplo 5.15:** Valor máximo y momento en que ocurre

In [3]:
print(tensiones['tension1'].max())

219.828


In [4]:
print(tensiones['tension1'].idxmax())

2021-02-20 00:00:00


**Ejemplo 5.16:** Categorización de valores

In [7]:
df_backup = tensiones.copy()  # recomendación antes de eliminar columnas


media = tensiones['tension1'].mean()
std = tensiones['tension1'].std()

def clasificar_voltaje(v):
    if v < media - std:
        return 'Bajo'
    elif v > media + std:
        return 'Alto'
    else:
        return 'Medio'

tensiones['Nivel Voltaje 1'] = tensiones['tension1'].apply(clasificar_voltaje)

print(tensiones)

                     tension1  tension2 Nivel Voltaje 1
fecha                                                  
2021-01-01 00:15:00   119.549   118.911           Medio
2021-01-01 00:30:00   119.498   118.218           Medio
2021-01-01 00:45:00   118.792   118.595           Medio
2021-01-01 01:00:00   119.179   118.334           Medio
2021-01-01 01:15:00   119.298   118.063           Medio
...                       ...       ...             ...
2021-06-26 22:45:00   119.719   119.071           Medio
2021-06-26 23:00:00   119.642   118.167           Medio
2021-06-26 23:15:00   119.388   118.764           Medio
2021-06-26 23:30:00   119.620   118.585           Medio
2021-06-26 23:45:00   119.388   118.958           Medio

[16974 rows x 3 columns]


**Ejemplo 5.17:** Clasificación con NumPy: `np.where()`

In [9]:
import numpy as np
tensiones['Voltaje Seguro 1'] = np.where(tensiones['tension1'] < 120, 'Sí', 'No')

## Ejercicios

**Ejercicio 5.9:**
 Para los datos contenidos en la columna tension2, determine:

 a) Valor máximo y fecha en que ocurre

In [ ]:
# Espacio para resolver el ejercicio

b) Repita el Ejemplo 5.16 para la columna tension2.

In [ ]:
# Espacio para resolver el ejercicio

c) Repita el Ejemplo 5.17 para la columna tension 2.

In [ ]:
# Espacio para resolver el ejercicio

**Ejercicio 5.10:** Establezca una columna que se llama "Reporte", en la cual se consigne si la tensión en las columnas tension1 y tension2 tuvieron un valor bajo 119. Sugerencia: Ver Ejemplo 5.5

In [ ]:
# Espacio para resolver el ejercicio

**Ejercicio 5.11:** Analice la diferencia que existe entre utilizar:

`print(tensiones['tension1'].mean())`

y


print(tensiones['tension1'].mean) [texto del vínculo](https://)

In [ ]:
# Espacio para resolver el ejercicio

**Ejercicio 5.12:**
 Para los datos contenidos en la columna tension2, determine:

 a) Valor mínimo y fecha en que ocurre